backend/sync-mounter-service/src/jvmMain/kotlin/services/MountService.kt

```diff
@@ -0,0 +1,102 @@
+package dk.sdu.cloud.sync.mounter.services
+
+import dk.sdu.cloud.calls.RPCException
+import dk.sdu.cloud.sync.mounter.SyncMounterConfiguration
+import dk.sdu.cloud.sync.mounter.api.*
+import io.ktor.http.*
+import java.io.File
+
+class MountService(
+    val config: SyncMounterConfiguration
+) {
+    private val MS_BIND: Long = 4096
+    private val O_NOFOLLOW = 0x20000
+
+    fun mount(request: MountRequest): MountResponse {
+        request.items.forEach { item ->
+            val source = File(joinPath(config.cephfsBaseMount ?: "/mnt/cephfs", item.path))
+            if (!source.exists() || !source.isDirectory) {
+                throw RPCException.fromStatusCode(HttpStatusCode.NotFound, "Invalid source")
+            }
+
+            val target = File(joinPath(config.syncBaseMount ?: "/mnt/sync", item.id))
+            if (target.exists()) {
+                unmount(UnmountRequest(listOf(MountFolderId(item.id))))
+            }
+
+            if (!target.mkdir()) {
```

Comment:
I think we need some input validation on the source and target. Make sure they are actually children of the folder you expect.

Reply:
We can check if `source` starts with `cephfsBase + "home"`, is this what you mean?
And in that case, `target` is already made from `syncBase + itemId`, what should we validate?

Reply:
You need to check that the _resolved_ source is inside the expected folder. The biggest problem is if the source service starts acting malicious and tries to mount arbitrary folders by using relative paths. For example, "$cephfsBase/home/foo/bar/../../../../../../something/else/entirely` still starts with the expected path but will resolve to something completely different.